packages/synapse-react-client/src/components/TableColumnSchemaEditor/Validators/ColumnModelValidator.test.ts

```diff
@@ -0,0 +1,664 @@
+import { ColumnTypeEnum } from '@sage-bionetworks/synapse-types'
+import { ColumnModelFormDataZodSchema } from './ColumnModelValidator'
+
+describe('ColumnModel validation', () => {
```

Comment:
So many tests to make sure this works reasonably. At a high level, check the following behavior
- Doesn't require fields that should be optional
- Respects max/min values for maximumSize, maximumListLength
- Converts data properly, such as
  - Drops empty strings
  - Handles data that may be numbers (returned from the API) or strings (entered by a form field)
  - Validates the contents `defaultValue` and `enumValues` fields based on the column type